### DISCUSSION TITLE
Aquarite wiring and usage instruction?

### DISCUSSION THREAD

**Discussion by magimat** [tagyoureit/nodejs-poolController#415] [RS485] [answered]
Hi,

I have an Aquarite salt generator and from what I understand, I should be able to communicate with it using this software and a RS485 connection, but I can't find any more specific info about it.

For the wiring, the manual indicates the 4 wire connection, so that should be it, but once connected, how do I communicate with it?  Can I read data and set like the output level from the software?

Any pointer to some more info would be appreciated.

thanks!

**Comment by magimat** [comment, marked as answer]
well I guess I just found what I was looking for:

https://github.com/tagyoureit/nodejs-poolController/wiki/Chlorinator

it was right there!

Now to order an RS485 usb device and do some testing.... this pool season should be really fun :)

**Reply by tagyoureit** [reply]
See https://github.com/tagyoureit/nodejs-poolController/issues/293.  You may need to supply 12v to the RS485 on the Aquarite to get it to talk.

**Reply by Hect0rG** [reply]
You should be able to find a 10 or 12V DC adapter to power it. A 10V adapter is hard to find and if you do find it, will probably be more expensive. I used a 12V adapter with a buck converter as @rstrouse recommended to me a little while back and adjust it to supply only 10V. The circuit on the Aquarite will work within the 7 to 12 volt range. If you go any lower than 7 you will not be able to communicate with the Aquarite. If you look at the print on the Aquarite it clearly states 10V so to be safe, I wouldn't go any higher. The circuit pulls .001 amps so any small DC adapter will work.

**Reply by magimat** [reply]
thanks!

**Comment by magimat** [comment]
finally had some time to do some real testing on that, so I connected everything accordingly, but I dont seem to get any activity from my SWG...

poolcontroller fires up fine. but it doest seem to connect in any way to my SWG

Since I am directly connecting pool-controller to my SWG and I dont have any real controller panel, is there some config I need to do? I read something about virtual controller somewhere, but I can't find any info about it in the wiki.

thanks!

PS: i get some occasionnal blink on the RX led of my RS485 dongle, so I am assuming everything is wired correctly, but I could be wrong!

**Reply by tagyoureit** [reply]
You need to tell Nixie you have a chlorinator.  Go to Systems (gear in upper right) -> Chemistry -> Add Controller -> Nixie.

![image](https://user-images.githubusercontent.com/1646840/168332086-200349c4-f976-4aa8-a03a-6cff98113946.png)

**Reply by magimat** [reply]
I see and this configuration is done through dashPanel I guess?  Because so far I only had the server running and was hoping to only use config file because all I want is the api, but I'll give dashPanel a shot to see if I can configure this

thanks

**Reply by tagyoureit** [reply]
Once you have the system configured, you can delete dP.

**Comment by magimat** [comment]
ok, got this far and I added a chlorinator, but still nothing as far as I can tell.... the RS485 connection in the UI shows as open, but no bytes are transferred. any idea how to debug this any further?

**Reply by rstrouse** [reply]
Yes upload a replay of your system that is about a minute long.  There should be at least failed send packets to the chlorinator.  The chlorinator will only respond when the pool body is on.

**Comment by magimat** [comment]
I’ll try that later, but I have a feeling my rs485 is not seeng any packet, I doubled checked wiring and checked I have 10v at the swg connector, but still, not getting any data.

I made sure serial port is disabled on my rpi, but the linux command from the wiki returns nothing :(

**Comment by rstrouse** [comment]
Here are some tips.  Certain versions of raspberry pi steal the serial port for bluetooth.  If you are using an adapter that connect to the serial port (non-USB) then you need to make sure you disable the serial console as that stupid interface steals the ttyS0 port.  If you are using a USB adapter then the port will be ttyUSB0 and can be affected by the bluetooth implementation.  Finally, swap your wires the traffic will read inverted and invalid but the port will open if they are reversed.

**Comment by magimat** [comment]
ok, I think disabling bluetooth did it because I finally see some change.

My SWG panel now has the remotely controlled led lit and I can see some activity in the RS485 panel in the UI

BUT....  I still can't actually control the SWG from the software, it shows as 1500 ppm and 0% output whatever I do... I'm trying to simply set the output to say 15%, but no luck,

Probably just missing some config steps at this point, but still... 

Thanks a lot for all the help, I have the feeling I'm almost there!

**Comment by magimat** [comment]
ok, I have some more updates to report.

It actually works now...  turns out I only needed to click on the pool button to enable it and now everything works and I am able to get info from my SWG and set the ouput % from the software

so all is good, but I still need to figure a few things out... turns out I'm not 100% sure bluetooth was my problem, I also started testing directly using "npm start" from console instead of using the docker image and it started working at this point and now I know it works, but not using docker.

When I try to start nodejs-poolController with docker, rs485 communication is dead and nothing works... but if I start poolController directly in the console, it works,

So, it means somehow, docker can't access my usb dongle. I tried about everything I know, --privileged, --group-add dialout, even -v /dev:/dev but nothing seems to works inside docker .... sad because I really wanted this to work inside a container and I did access some usb devices from docker on the pi before, so it must be possible

Anyway idea about that??

Anyway, thanks a lot for all the help, at least now I can make it work

**Comment by magimat** [comment]
as a follow up, I can now confirm that my problem was docker from the beginning :(

I juste re-enabled bluetooth and it still works, just not inside docker

**Comment by magimat** [comment]
one last update, I finally have it working inside docker!!

I resorted to build my own docker image natively on the raspberry pi, simply using a fresh clone of the repo, but apparently, building the image directly on the pi did have some effect, because now it work!

So there is still some docker magic that I can't explain, but now it works, so I'm a happy camper :)

**Reply by tagyoureit** [reply]
Not sure if you saw the [docker wiki](https://github.com/tagyoureit/nodejs-poolController/wiki/Docker).  Let us know if something there needs to be changed.  I'm not an expert on Docker - those were provided by a few community members - but they've been very helpful for others.

**Reply by magimat** [reply]
yea, the info about docker in the wiki is fine, maybe just add somewhere that if all else fails, one could try to build the docker image directly on the host where it will run, at least this is what fixed my issue.

**Comment by magimat** [comment]
maybe one last question. Now that everything works, my goal is to only use the rest API to control the SWG and get some basic info,

I can set the output and get salt level ppm just fine, but I was also hoping to be able to get the status info from the SWG. From what I can read in the wiki, it should be available (https://github.com/tagyoureit/nodejs-poolController/wiki/Chlorinator)

I'm looking for 

Error bit: 0=Ok, 1=No Flow, 2=Low Salt, 4=High Salt, 144=Clean Salt Cell

for example, if I turn the main pump off, the no flow led is lit on the panel and I was hoping to be able to pick up this info through the api.
